tests/stub/types/test_temporal_types.py

```diff
@@ -35,54 +35,82 @@ def _create_direct_driver(self):
+        dt = types.CypherDateTime(2022, 6, 7, 11, 52, 5, 0, utc_offset_s=7200)
+        result = self._session.run("RETURN $dt AS dt", params={"dt": dt})
+        records = list(result)
+        self.assertEqual(len(records), 1)
+        self.assertEqual(len(records[0].values), 1)
+        self.assertEqual(dt, records[0].values[0])
+
+    def _test_zoned_date_time(self, patched):
+        script = "echo_zoned_date_time.script"
+        if patched:
+            script = "echo_zoned_date_time_patched.script"
+        self._start_server(script)
         self._create_direct_driver()
         self._session = self._driver.session("w")
-        result = self._session.run("RETURN $dt AS dt", params={
-            "dt": types.CypherDateTime(
-                2022, 6, 7, 11, 52, 5, 0,
-                utc_offset_s=7200, timezone_id="Europe/Stockholm"
-            )
-        })
-        list(result)
-
-
-class _TestTemporalTypesPatchedBolt(_TestTemporalTypes):
-
-    @driver_feature(types.Feature.BOLT_PATCH_UTC)
-    def _test_date_time_with_patch(self):
-        self._start_server("echo_date_time_patched.script")
+        dt = types.CypherDateTime(
+            2022, 6, 7, 11, 52, 5, 0,
+            utc_offset_s=7200, timezone_id="Europe/Stockholm"
+        )
+        result = self._session.run("RETURN $dt AS dt", params={"dt": dt})
+        records = list(result)
+        self.assertEqual(len(records), 1)
+        self.assertEqual(len(records[0].values), 1)
+        self.assertEqual(dt, records[0].values[0])
+
+    def _test_unknown_zoned_date_time(self, patched):
+        tx_count = 0
+
+        def work(tx):
+            nonlocal tx_count
+            tx_count += 1
+            res = tx.run("MATCH (n:State) RETURN n.founded AS dt")
+            with self.assertRaises(types.DriverError) as exc:
+                res.next()
+            self.assertIn("Europe/Neo4j", exc.exception.msg)
+            raise exc.exception
+
+        script = "echo_unknown_zoned_date_time.script"
+        if patched:
+            script = "echo_unknown_zoned_date_time_patched.script"
+        self._start_server(script)
         self._create_direct_driver()
         self._session = self._driver.session("w")
-        result = self._session.run("RETURN $dt AS dt", params={
-            "dt": types.CypherDateTime(2022, 6, 7, 11, 52, 5, 0,
-                                       utc_offset_s=7200)
-        })
-        list(result)
+        with self.assertRaises(types.DriverError):
+            self._session.read_transaction(work)
+        self._server.done()
+        self.assertEqual(tx_count, 1)
+
+    def _test_unknown_then_known_zoned_date_time(self, patched):
+        def work(tx):
+            res = tx.run("MATCH (n:State) RETURN n.founded AS dt")
+            with self.assertRaises(types.DriverError) as exc:
+                res.next()
+            self.assertIn("Europe/Neo4j", exc.exception.msg)
+
+            rec = res.next()
```

Comment:
Should we not expect result to be in invalid state by this point given the error above?

Reply:
I figured it would be good to be as lenient as possible. Just because we can't parse a certain record, doesn't mean we can't try to parse the next record. If a user wants to read "all records the driver can possible understand" they'd have to first know which those are and filter them out somehow with cypher if the driver didn't provide a way to continue reading the record stream after such a hickup.

Reply:
Java driver currently follows the logic of failing when it has problems with unpacking types. This is not exclusive to datetime. If we want to change this, we should probably discuss it as a unification thing.

Reply:
Are we on the same page @thelonelyvulpes ? I'm talking about the scenario where the driver receives a zone id from the server that's unknown to the driver. What could the server possibly do about this or even know about this in advance?

Reply:
In my view, if `Result` throws on record consumption, it should no longer be considered to be in a valid state. One option that we could consider is to throw on accessing the actual value.

Reply:
In my view, if `Result` throws on record streaming, it should no longer be seen as a valid handle. One option that we could consider is to throw on accessing the actual value.

Reply:
Summary of off-GH discussion: for 4.4 it'll be what ever is non-breaking. For instance, for Java this will mean break the Result, for Python keep it usable. For 5.0 it should keep the Result object in tact. Throwing when actually accessing the property/attribute in question is a good option here.